.github/workflows/main.yml

```
@@ -258,4 +258,4 @@ jobs:
     steps:
     - uses: actions/checkout@v4
     - uses: taiki-e/install-action@cargo-hack
-    - run: cargo hack check --all-targets --rust-version --workspace --ignore-private
+    - run: cargo hack check --all-targets --rust-version --workspace --no-private
```

```suggestion
    - run: cargo hack check --rust-version --workspace --no-private --no-dev-deps
```

https://github.com/rust-lang/cargo/pull/13254#issuecomment-1895977917

> One thing to note is that the use of the --all-targets flag (that builds tests/examples) needed to be removed and the --no-dev-deps flag is needed. This is because the dev-dependencies use versionless manifests that cannot be read by the old compiler.
> 
> That said, we are not running tests with the old compiler, so I don't see a problem if the test suite cannot be built on the old compiler.

---

How I missed that. Thank you anyhow :)